src/optimagic/visualization/backends.py

```diff
@@ -25,6 +27,26 @@ def __call__(
         legend_properties: dict[str, Any] | None,
         margin_properties: dict[str, Any] | None,
         horizontal_line: float | None,
+        subplot: Any | None = None,
```

Comment:
I like this `subplot` argument and how you use it a lot! Can you make sure that it is clear how to use it. That is, here in the protocol docstring you should explain it on an abstract level, and then I would expect to see a description of the argument in the docstring of the backend-specific functions. The docstrings of the backend-specific functions do not need to repeat all of the argument descriptions that are the same across backends, but only the ones that differ between backends.